src/components/selectionBoxLayer.ts

```diff
@@ -0,0 +1,116 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Component {
+  export class SelectionBoxLayer extends AbstractComponent {
+    private _box: D3.Selection;
+    private _core: D3.Selection;
+    private _edgeT: D3.Selection;
+    private _edgeB: D3.Selection;
+    private _edgeL: D3.Selection;
+    private _edgeR: D3.Selection;
+    private _cornerTL: D3.Selection;
+    private _cornerTR: D3.Selection;
+    private _cornerBL: D3.Selection;
+    private _cornerBR: D3.Selection;
+
+    private _topLeft: Point = { x: 0, y: 0 };
+    private _bottomRight: Point = { x: 0, y: 0 };
+
+    private _detectionRadius = 2;
```

Comment:
As said from the other PR, if we want to properly decouple interactions from the component itself, I'm unsure about having this variable here.  That said, we might just need to rename this variable.